tests/unit/providers/weather/backends/test_accuweather.py

```diff
@@ -553,9 +615,46 @@ async def test_get_location(
     assert location == expected_location
 
 
+@pytest.mark.asyncio
+async def test_get_location_from_cache(
+    mocker: MockerFixture,
+    accuweather_parameters: dict[str, Any],
+    accuweather_location_response: bytes,
+    response_header,
+) -> None:
+    """Test that we can get the location from cache."""
+    redis_mock = mocker.AsyncMock(spec=Redis)
+
+    async def mock_get(key) -> Any:
+        return accuweather_location_response
+
+    redis_mock.get.side_effect = mock_get
+
+    expected_location: AccuweatherLocation = AccuweatherLocation(
+        key="39376_PC", localized_name="San Francisco"
+    )
+    country: str = "US"
+    postal_code: str = "94105"
+    mock_client: Any = mocker.AsyncMock(spec=AsyncClient)
+
+    accuweather = AccuweatherBackend(cache=redis_mock, **accuweather_parameters)
+    location: Optional[AccuweatherLocation] = await accuweather.get_location(
+        mock_client, country, postal_code
+    )
+
+    assert location == expected_location
+    redis_mock.get.assert_called_once_with(
+        f"AccuweatherBackend:v1:/locations/v1/postalcodes/{country}/search.json:"
+        f"{hashlib.blake2s(str({'q': postal_code}).encode('utf-8')).hexdigest()}"
+    )
+    assert not mock_client.get.called
```

Comment:
```suggestion
    mock_client.get.assert_not_called()
```

suggestion: `assert_not_called` has a nicer error message with details (mock name, number of times it was called) if the assertion fails.